### ISSUE TITLE
don't interpret underscores to mean 'italic' when used in the middle of a word.

### ISSUE THREAD

**Issue by raboof** [vector-im/riot-web#7368]
### Description

When displaying a string that contains underscores, such as `#_ircnet_#hsnijmegen:irc.snt.utwente.nl`, the underscores are removed and the 'ircnet' fragment is shown in italic ("#_ircnet_#hsnijmegen:irc.snt.utwente.nl").

It would be nicer to leave underscores be when encountered in the middle of a word, and only convert them to italic when they are added on word boundaries.

### Version information

- **Platform**: web (in-browser)

For the web app:

- **Browser**: Firefox 61.0.1
- **OS**: Debian Linux
- **URL**: riot.im/app

**Comment by t3chguy** [comment]
This is the behaviour of CommonMark

Related https://github.com/vector-im/riot-web/issues/6434 https://github.com/vector-im/riot-web/issues/3534

**Comment by ara4n** [comment]
dup https://github.com/vector-im/riot-web/issues/4674

**Comment by raboof** [comment]
I'm not sure this is really a duplicate of #4674 because that is specifically about URL's and this is a non-URL string. A fix for #4674 may or may not fix this case.

**Comment by ara4n** [comment]
Yup, you're right - it's actually a regression of https://github.com/vector-im/riot-web/issues/3428. It's probably very similar to #4674 but may not be. Will reopen.

**Comment by lampholder** [comment]
Right, so commonmark (I believe) does the 'right thing' for underscores in the middle of words.

The problem is, it looks like commonmark doesn't consider # a word character: [reference impl of commonmark showing the difference](https://spec.commonmark.org/dingus/?text=%23_ircnet_%23hsnijmegen%3Airc.snt.utwente.nl%0A%0Ah_ircnet_%23hsnijmegen%3Airc.snt.utwente.nl%0A).

Fwiw, GFM has exactly the same issue: #_ircnet_#hsnijmegen:irc.snt.utwente.nl

https://github.com/vector-im/riot-web/issues/4674#issuecomment-355850720 outlines some possible solutions. I'd really like us to stick to an agreed markdown standard; could we address this with some creative pilling (by which I mean turning all aliases to pills and then mandating our own formatting?)

**Comment by raboof** [comment]
What are/is pills/pilling?
